README.md

```diff
@@ -1,125 +1,148 @@
-2019/06/27 08:57:23 [INFO] starting ingressListener watcher
+  -cluster-name-annotation string
+        Annotation that will mark a pod as part of a cluster (default "cluster-name.envoy.uw.io")
+  -log-level string
+        log level trace|debug|info|warning|error|fatal|panic (default "warning")
+  -sources string
+        (Required) Path of the config file that keeps static sources configuration
 ```
 
-Start a local instance of envoy that uses the control plane
-```
-$ make local-envoy
-$ make local-envoy-mac # on macOS
-```
+### Sources configuration - control plane
 
-You can inspect the current Envoy config by visting `/config_dump`
-```
-$ curl http://localhost:9901/config_dump
-```
+Each source is a kubernets cluster that the control plane is going to watch events from. An example config will be:
 
-Create a pod in the cluster 'test-cluster', an `IngressListener` for the app 'test-app' and an `EgressListener` that allows
-'test-app' to query `test-cluster:8080` by making requests to envoy at `127.0.0.1:9090`.
-```
-$ cd manifests/examples
-$ kubectl --context=exp-1-aws -n labs apply -f .
 ```
+    [
+      {
+        "name": "aws",
+        "kubeconfig": "/etc/aws/aws.kubeconfig",
+        "listenersource": true
+      },
+      {
+        "name": "gcp",
+        "kubeconfig": "/etc/gcp/gcp.kubeconfig",
+        "listenersource": false
+      }
+    ]
 
-You should see events registered in the logs of the control plane
-```
-2019/06/27 09:16:45 [DEBUG] received ADDED event for cluster test-cluster ip: 10.2.7.139
-```
-```
-2019/06/27 09:20:45 [DEBUG] received ADDED event for ingress listener example-listener: 0.0.0.0:8080 -> 127.0.0.1:8081
-2019/06/27 09:20:45 [DEBUG] Updating snapshot for node test-app
-2019/06/27 09:20:45 [DEBUG] Response: responding (2) with type: type.googleapis.com/envoy.api.v2.Cluster, version: 2019-06-27 01:20:45.705178 +0100 BST m=+240.565251561, resources: 1
-2019/06/27 09:20:45 [DEBUG] Response: responding (1) with type: type.googleapis.com/envoy.api.v2.Listener, version: 2019-06-27 01:20:45.705185 +0100 BST m=+240.565257900, resources: 1
 ```
 
-Check `http://localhost:9901/config_dump` to see the listeners and clusters that have been added to Envoy by the control plane.
+You need to supply a list of source and the location of the respective kubeconfig files.
+`listenersource` defines the primary kubernetes cluster, meaning the cluster which the control plane will watch for listeners. It should be set to true only for 1 cluster, the same one that the control plane is deployed. That way envoy sidecars in each kubernetes cluster will talk to the local control plane to fetch config, listeners will only be defined in the same cluster where the envoy sidecar is running and clusters will be generated from all the available sources.
+Losing networking between clusters means that remote cluster members will disappear but everything will keep working for the local cluster.
```

Comment:
If we don't already, could we create a check in the code to make sure there is only one 'primary' kubernetes cluster?

Reply:
Rethinking this one, I believe we should allow as many `listenersource` cluster as the user wants. It makes sense for how we deploy to have only one, but I can imagine setups where there you run the control plane in one cluster and others fetch listener configuration from there. Let's leave it until it becomes an issue??

Reply:
:+1: